snapcraft/tests/test_lifecycle.py

```diff
@@ -656,8 +657,42 @@ def test_prime_with_build_info_records_snapcraft_yaml(self):
     plugin: nil
     prime: []
     stage: []
-architectures: [amd64]
+    stage-packages: []
+architectures: [{}]
+""".format(self.project_options.deb_arch))
+        self.assertThat(
+            os.path.join('prime', 'snap', 'snapcraft.yaml'),
+            FileContains(expected))
+
+    def test_prime_with_stage_packages(self):
+        self.useFixture(fixtures.EnvironmentVariable(
+            'SNAPCRAFT_BUILD_INFO', '1'))
+        self.useFixture(fixture_setup.FakeAptCache([
+            ('test-package1', 'test-version1'),
+            ('test-package2', 'test-version2')]))
+
+        self.make_snapcraft_yaml("""parts:
+  test-part:
+    plugin: nil
+    stage-packages: [test-package1=test-version1, test-package2]
 """)
+
+        lifecycle.execute('prime', self.project_options)
+
+        expected = ("""name: test
+version: 0
+summary: test
+description: test
+confinement: strict
+grade: stable
+parts:
+  test-part:
+    plugin: nil
+    prime: []
+    stage: []
+    stage-packages: [test-package1=test-version1, test-package2=test-version2]
```

Comment:
The order in the test results is reversed. By the looks of it, FakeAptCache will have an undefined order since it's using a dict, as is the YAML that is put in a dict by snapcraft internally. I think you should be going through the packages in the test and checking that they're in there without relying on the order.